platforms/azure/giantnetes/main.tf

```diff
@@ -244,14 +244,26 @@ module "worker" {
   vm_size               = "${var.worker_vm_size}"
 }
 
-module "vpn" {
+module "vpn" "vpn_connection_0" {
   source = "../../../modules/azure/vpn"
 
   cluster_name              = "${var.cluster_name}"
   location                  = "${var.azure_location}"
   resource_group_name       = "${module.resource_group.name}"
   subnet_id                 = "${module.vnet.vpn_subnet_id}"
   vpn_enabled               = "${var.vpn_enabled}"
-  vpn_right_gateway_address = "${var.vpn_right_gateway_address}"
-  vpn_right_subnet_cidr     = "${var.vpn_right_subnet_cidr}"
+  vpn_right_gateway_address = "${var.vpn_right_gateway_address_0}"
+  vpn_right_subnet_cidr     = "${var.vpn_right_subnet_cidr_0}"
+}
+
+module "vpn" "vpn_connection_1" {
```

Comment:
We either:
1. call module twice (like here) and module uses some suffixes for resources it creates. But we need to remove hard coding with `_0` `_1` from inside module (module does not know that there are two VPNs at the same time).
2. or just call module once and always have two gateways (module by default creates two vpn gateways)

I like the first one as this is actual purpose of modules.

Currently you have this knowledge (about two vpns) both inside module and when calling module.